add a Dockerfile

jcupitt opened the pull request:
It seems to work for me ... test with:

	docker build \
		--build-arg VCS_REF=`git rev-parse --short HEAD` \
		-t <user>/structural-pipeline:latest .

Fetching the FSL key seems a little unreliable, I'm not sure why. If you
get an error, try just running again.

see https://github.com/DevelopingHCP/structural-pipeline/issues/2

jcupitt committed:
add a Dockerfile

It seems to work for me ... test with:

	docker build \
		--build-arg VCS_REF=`git rev-parse --short HEAD` \
		-t <user>/structural-pipeline:latest .

Fetching the FSL key seems a little unreliable, I'm not sure why. If you
get an error, try just running again.

see https://github.com/DevelopingHCP/structural-pipeline/issues/2

jcupitt commented:
(Dockerfile shamelessly copied from MIRTK)

schuhschuh commented:
Sweet! BTW I myself have no permissions to merge. Either wait for @amakropoulos's reply or get permissions from in-house if your part of BioMedIA/dHCP.

jcupitt committed:
add notes to README

I made it urge people to do a build in docker, perhaps that's not wise

jcupitt commented:
I added some notes to the README. It should probably use `-v` to attach the user's data files to the container for processing, I'll experiment.

jcupitt committed:
README polish

jcupitt committed:
improve notes on running the pipeline

jcupitt committed:
update pipeline script for FSL 5 prefix

fsl5 has "fsl5.0-" in front of all exe names

the dockerfile needs to get root shell to run the fsl init script

jcupitt committed:
can now run the pipeline successfully

jcupitt commented:
It seems to be working well now. I'm going to merge to master on my fork, I'll leave this here.

The README has some notes -- tldr: build the pipeline to a docker image with:

```
# docker build -t <user>/structural-pipeline:latest .
```

Run the pipeline on the file `sub-CC00183XX11_ses-60300_T2w.nii.gz` in the subdirectory `data` of the current directory:

```
# docker run --rm -t -v $PWD/data:/data \
    <user>/structural-pipeline:latest \
    bash -c ". /etc/fsl/fsl.sh; \
        cd /usr/src/structural-pipeline; \
        ./dhcp-pipeline.sh subject1 session1 44 \
            -d /data -T2 /data/sub-CC00183XX11_ses-60300_T2w.nii.gz -t 8"
```

I'll try and get OpenMOLE using this next.

jcupitt committed:
add note about -u argument to docker

jcupitt committed:
tweak README
